tests/Microsoft.Identity.Web.Test/AuthorityHelpersTests.cs

```diff
@@ -76,10 +76,11 @@ public void BuildAuthority_CiamAuthority_ReturnsValidAadAuthority()
             };
             string expectedResult = options.Authority + TestConstants.Domain;
 
-            string? result = AuthorityHelpers.BuildCiamAuthorityIfNeeded(options.Authority);
+            string? result = AuthorityHelpers.BuildCiamAuthorityIfNeeded(options.Authority, out bool preserveAuthority);
 
             Assert.NotNull(result);
             Assert.Equal(expectedResult, result);
+            Assert.False(preserveAuthority);
         }
 
         [Fact]
```

Comment:
Is there any way to get some coverage over the changes in  UpdateMergedOptionsFromMicrosoftIdentityOptions ?

Reply:
Yes, I would just need some support in understanding what the different cases look like for CIAM CUD

Reply:
CIAM (non-CUD) authority can be one of the following: 

- https://westin.ciamlogin.com
- https://westin.ciamlogin.com/{GUID} (the GUID is the tenant id)
- https://westin.ciamlogin.com/westin.onmicrosoft.com  (display name of the tenant)

CIAM CUD authority will be: 

- https://westin.com/{GUID}/v2.0

Reply:
Not sure this helps, happy to chat directly about this if needed.